### PR TITLE
Fix Deprecation/Future Warnings in Notebook 211-Speech-to-Text

### PR DESCRIPTION
**Pull request by YDX-2147483647**
- [`waveplot` is _deprecated_ in favor of `waveshow` since librosa v0.8.1](https://librosa.org/doc/latest/changelog.html?highlight=waveplot#v0-8-1), and [_removed_ in v0.9.0](https://librosa.org/doc/latest/changelog.html?highlight=waveplot#v0-9-0).

    [This was mentioned in the original pull request in Dec. 2021](https://github.com/openvinotoolkit/openvino_notebooks/pull/271#pullrequestreview-831901849), but not fixed (for unknown reason).

  Note that `max_sr` in [`waveplot`](http://librosa.org/doc/0.8.1/generated/librosa.display.waveplot.html?highlight=max_sr#librosa.display.waveplot) is removed in [`waveshow`](http://librosa.org/doc/0.9.2/generated/librosa.display.waveshow.html). `max_sr` controls how signals are down-sampled if they’re too long. (`waveshow` uses “a downsampled amplitude envelope” instead)

  > In the committed version, Imports are at the top of the notebook.

- `librosa.filters.mel` in `audio_to_mel`

  _FutureWarning_: Pass `sr=16000, n_fft=512` as keyword args. From version 0.10 passing these as positional arguments will result in an error.

#### Question

[The original pull requeset](https://github.com/openvinotoolkit/openvino_notebooks/pull/271) did NOT add [librosa](http://librosa.org/doc/latest/) (the audio analysis package used here) into `requirements.txt` or `.docker/Pipfile`. Is it on purpose? Should I tell [how to install it](http://librosa.org/doc/latest/install.html) in the notebook?

### CODE REVIEW

**Comment by YDX-2147483647** [comment]
Sorry, it's mine. `max_points` in `waveshow` should be an `int`, not `float`.

(I checked it on a private repo and manually copy the changes to this repo - the screenshot was right.)

**Comment by Debskij** [comment]
Hi @YDX-2147483647, thanks for your contribution. It would be great if you can add `librosa` to both `requirements.txt` and `Pipfile` (and lock it afterward).

**Comment by YDX-2147483647** [comment]
New discovery: librosa v0.8.1 exists in [`Pipfile.lock`](https://github.com/openvinotoolkit/openvino_notebooks/blob/6e76762887298565369ddb8cc08ac28ff5688ceb/.docker/Pipfile.lock#L1397-L1404).

I `pip install` again and found it required by ppgan v2.1.0.

```powershell
> pip show librosa
Name: librosa
Version: 0.8.1
……
Requires: audioread, decorator, joblib, numba, numpy, packag
ing, pooch, resampy, scikit-learn, scipy, soundfile
Required-by: ppgan
```

https://github.com/openvinotoolkit/openvino_notebooks/blob/6e76762887298565369ddb8cc08ac28ff5688ceb/requirements.txt#L26

[ppgan's `requirements.txt`](https://github.com/PaddlePaddle/PaddleGAN/blob/91dcc906fb8fe4a378c10de6b4c0d8525b1176a7/requirements.txt#L8):

```
librosa==0.8.1
```

**Comment by adrianboguszewski** [comment]
@YDX-2147483647 please add it to requirements and Pipfile anyway

**Comment by YDX-2147483647** [comment]
✓ OK, but maybe tomorrow.

**Review by adrianboguszewski — APPROVED** [review]

**Comment by Debskij** [comment]
Hi @YDX-2147483647. Thanks for updating `requirements.txt` and `Pipfile`, still I can see that you haven't updated the `Pipfile.lock` file. To do that you have to (on your own machine) get into the directory `openvino_notebooks/.docker` and run command `pipenv lock`. Please add the updated file to the repository when it is ready.

**Comment by YDX-2147483647** [comment]
Uhh… Thanks for helping me. In fact I had noticed “(and lock it afterward)”, but got stuck at `pipenv install` for hours (even on `main`). I am working on it.

**Comment by YDX-2147483647** [comment]
Could I leave `Pipfile.lock` as it is?

**Reason**: As mentioned above, there’s no new requirement at all — we only elevate `librosa` from `ppgan`'s sub-dependency to a direct dependency. Current `Pipfile.lock` does contain `librosa` (v0.8.1), making it a valid environment.

**My difficulty**: I can’t `pipenv install` even on main because of PyTorch.

<details>
    <summary>My tries</summary>
    <p>(Failures omitted)</p>
    <ul>
        <li>
            <p>HTTP Errors.</p>
            <p>I use a mirror source instead, then manually change <code>Pipfile</code>s back before I commit.</p>
        </li>
        <li>
            <p>PyPI can't recognize <code>+cpu</code> for <code>torch</code> and <code>torchvision</code>.</p>
            <p>
                With everything else settled, I <code>pipenv run pip install</code> from <a href='https://download.pytorch.org/whl/'>PyTorch's site</a>.
            </p>
        </li>
        <li>
            <p><code>pipenv lock</code> failed. (<code>pipenv lock</code> without PyTorch succeeded.)</p>
            <p>
                Use <a href='https://pipenv.pypa.io/en/latest/advanced/#specifying-package-indexes'>advanced syntax</a> in <code>Pipfile</code>.
            </p>
        </li>
        <li>
            <p>No space left on device.</p>
           	<p>
                Clean pip cache. Used up again. Clean. Again. …
            </p>
            <p>
                (Maybe <a href='https://github.com/pypa/pipenv/issues/4963'>a <code>pipenv</code> issue</a>: Pipenv downloads PyTorch for all versions of Python, grabbing 16GB of data instead of just 1.7GB)
            </p>
        </li>
    </ul>
    <p>
        It sounds ridiculous that I am fighting against PyTorch environment days and night without using it… But it's truth. <code>╮(╯_╰)╭</code>
    </p>
</details>

File: .docker/Pipfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ jupyter-nbrequirements = "*"
 jupyterhub = "==2.2.*"
 jupyterlab = "*"
 jupyterlab-git = "==0.30"
+librosa = ">=0.8.1"
```
Comment (adrianboguszewski) [inline .docker/Pipfile:20]:
```suggestion
```

File: requirements.txt [hunk 2]
```diff
@@ -25,6 +25,9 @@ paddlehub
 paddle2onnx>=0.6, <=0.9.6
 ppgan==2.1.*
 
+# Audio analysis requirements
+librosa>=0.8.1
+
```
Comment (adrianboguszewski) [inline requirements.txt:30]:
```suggestion
```

**Comment by adrianboguszewski** [comment]
@YDX-2147483647 ok, we will add librosa in another PR, let's merge your changes in 211

**Comment by YDX-2147483647** [comment]
Thanks!

**Review by Debskij — APPROVED** [review]

**Pull request merged**